session/session.go

```diff
@@ -1036,6 +1037,18 @@ func (s *session) executeStatement(ctx context.Context, connID uint64, stmtNode
 }
 
 func (s *session) Execute(ctx context.Context, sql string) (recordSets []sqlexec.RecordSet, err error) {
+	defer func() {
+		r := recover()
+		if r == nil {
+			return
+		}
+		if str, ok := r.(string); !ok || !strings.HasPrefix(str, memory.PanicMemoryExceed) {
```

Comment:
I think a better place is `ExecStmt.Exec()` function, or `ExecStmt.handleNoDelay()` and `ExecStmt.handlePessimisticSelectForUpdate()` function.

Reply:
done.